### ISSUE TITLE
Breaking changes in webpack v2.1.0-beta.23

### ISSUE THREAD

**Issue by IAMtheIAM** [shakacode/sass-resources-loader#13]
Bug:

Webpack v2.1.0-beta.23 no longer allows custom properties on the webpack config object, breaking this plugin since it uses 'sass-resources' on the object. Please update the plugin to be compatible with the upcoming version of webpack 2. Thanks!

```
ERROR in ./src/app-components/app/app.style.scss
Module build failed: ModuleBuildError: Module build failed: TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.resolve (path.js:184:7)
    at C:\Source\Atlas\node_modules\sass-resources-loader\lib\loader.js:64:31
    at Array.map (native)
    at Object.module.exports (C:\Source\Atlas\node_modules\sass-resources-loader\lib\loader.js:63:33)
    at C:\Source\Atlas\node_modules\webpack\lib\NormalModule.js:143:35
    at C:\Source\Atlas\node_modules\loader-runner\lib\LoaderRunner.js:359:11
    at C:\Source\Atlas\node_modules\loader-runner\lib\LoaderRunner.js:225:18
    at runSyncOrAsync (C:\Source\Atlas\node_modules\loader-runner\lib\LoaderRunner.js:138:3)
    at iterateNormalLoaders (C:\Source\Atlas\node_modules\loader-runner\lib\LoaderRunner.js:224:2)
    at C:\Source\Atlas\node_modules\loader-runner\lib\LoaderRunner.js:197:4
    at Storage.finished (C:\Source\Atlas\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:39:16)
    at C:\Source\Atlas\node_modules\enhanced-resolve\node_modules\graceful-fs\graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:413:3)
```

See: https://github.com/webpack/webpack/releases/tag/v2.1.0-beta.23

**Comment by justin808** [comment]
@IAMtheIAM Any chance that you can submit a PR, and ideally have another github member verify that the changes work, and I'll review, merge, and push.

Plus, will this break older versions of webpack, such as v1 and older v2?

**Comment by IAMtheIAM** [comment]
I found the cause of the error. `webpack.options.context` is now undefined, it must be passed explicitly to the loader using the LoaderOptionsPlugin

      new webpack.LoaderOptionsPlugin({
         options: {
            sassResources: ['./src/assets/styles/variables.scss', './src/assets/styles/mixins.scss'],
            context: path.resolve(__dirname, '..') // must evaluate to root of project
         }
      }),

Now it works fine with this. But using current webpack config results in error. I'd like to submit a pull request but I'll have to figure out how to extract the `context` info from within the loader, otherwise the user has to pass the `context` to the loader explicitly.

**Comment by andregoncalvesdev** [comment]
that fixed for me @IAMtheIAM . using webpack 2.1.0-beta.25

**Comment by web2style** [comment]
Actually when learning how to migrate to the new version of webpack I figured out sass-resources is not needed at all in my case. This:


```
    new webpack.LoaderOptionsPlugin({
      options: {
        sassLoader: {
          data: '@import "app/styles/settings/_settings.scss";',
          includePaths: 'app/styles'
        },
        context: path.resolve(__dirname, '../../') // must evaluate to root of project
      }
```

Will do the same as this:

```
new webpack.LoaderOptionsPlugin({
      options: {
        sassLoader: {
          includePaths: 'app/styles'
        },
        sassResources: [
          'app/styles/settings/_settings.scss'
        ],
        context: path.resolve(__dirname, '../../') // must evaluate to root of project
      }
    }),
```

May be will be useful for smb else.

**Comment by IAMtheIAM** [comment]
@web2style Very nice, I will try that out as well, thanks for sharing.

**Comment by justin808** [comment]
I pushed up a beta: 1.2.0-beta.1.

@IAMtheIAM @web2style @andregoncalvesdev Any chance of a review: https://github.com/shakacode/sass-resources-loader/pull/16 and try out the beta?

**Comment by alexfedoseev** [comment]
Fixed in `1.2.0`.
